### DISCUSSION TITLE
Install imagemagick on Windows with only CLI

### DISCUSSION THREAD

**Discussion by jfoclpf** [ImageMagick/ImageMagick#3948] [Help]
I need imagemagick on my nodeJS project and I implemented a [Github Continuos Integration test suite in which imagemagick is installed](https://github.com/jfoclpf/autocosts/blob/master/.github/workflows/node.js.yml) before the test is performed. My Github CI test suite includes ubuntu, macos and windows.

For ubuntu and macos the installation of imagemagick is quite straightforward via `sudo apt` or `brew` respectively, but for Windows I'm lost on which commands shall I run on the CLI to install imagemagick.

Is there a way to install imagemagick on a Windows machine with only CLI (without prompting stuff or opening installations dialogues)?

**Comment by snibgo** [comment]
The Windows installation programs (such as ImageMagick-7.0.10-45-Q16-HDRI-x64-dll.exe) can be run with parameters. Run it with parameter "/?" to see what options can be used.

The options seem to be generic, not specific to ImageMagick, so for example there is no option for installing legacy components.

A developer may comment.

**Comment by jfoclpf** [comment]
Thanks
I am trying with chocolately

```
choco install imagemagick.app
```

BTW, sorry to ask, how do you download that imagemagick.exe file on Windows with CLI?

**Comment by snibgo** [comment]
I know nothing about chocolately.

IM Windows installation files are at https://imagemagick.org/script/download.php#windows

**Reply by jfoclpf** [reply]
Thank you @snibgo 
I already could install imagemagick on Windows but I'm facing the convert issue (conflict with the system convert exe).

Do you know how to impose the convert from imagemagick over the one of system windows?

**Comment by snibgo** [comment]
"convert.exe" is a v6 IM program. I do not recommend that new installations use IM v6, or IM v7 using v6 syntax by using "magick convert". Instead, I recommend using IM v7, with the "magick" command. V7 syntax is far better than v6.

If, for some reason, you really must use IM v6 "convert.exe", then you have a number of choices:

1. Put the IM directory in front of Windows directories on the search path. Then "convert" at the command line will always get the IM program, not the Microsoft program.

2. Rename the IM program, eg to "imconvert.exe".

3. Always use a directory prefix. For example, when I want to run convert, I type "%IM%convert", where %IM% is an environment variable that points to the IM directory.

**Comment by jfoclpf** [comment]
Thank you so very much. Yes, I really must use the IM v6 convert.exe and I
must implement option 1 (the third party npm package I am using runs the IM
convert command).

Do you know how to implement option 1 with windows Power Shell?

Thanks

**Comment by snibgo** [comment]
I don't use Powershell. I suppose it uses the same system path as other shells. This can be set (for Windows 8.1) on Control panel, System, Advanced, Environment variables, System variables, PATH.

**Comment by jfoclpf** [comment]
Thanks but I'm using a Continuous Integration test suite. I can only use CLI

**Comment by snibgo** [comment]
It can be set temporarily, for one console, with "set PATH=...". I don't know how to set it system-wide and permanently from the command line. Perhaps a registry setting.
